---
title: "The Gene Ontology: A Resource for Annotation and Analysis"
author: "Alex Sanchez (asanchez@ub.edu)"
date: "9th September 2016"
output: slidy_presentation
css: my.css
footer: "GO 4 Annotation and Analysis"
keep_md: true
---


# Introduction and Motivation

- A common problem in biological research is the study of lists of genes -or related entities like proteins, microRNAs, metabolites- that have been selected as characteristic in some situation.
    + The example list #1 has been obtained by comparing gene expression in healthy or affected (BC) breast tissue. That is these are the genes that express differently in cancer and non-cancer cells.
    + The example list #2 has been obtained by comparing gene expression in liver cells before and after a drug has been administered.

# What does this all mean?

- It is difficult that, simply by looking at the gene lists, any biologist -even an expert in the field- can gain much insight on what it means.
    + To help on this, genes are _annotated_ in different databases and 
    + These annotations are/can/may be used to (try to) provide biological significance to the gene lists.
- This is what this talk/lab is about.

<!-- Posar un exemple -->

# Outline of the talk

- Background
    + Biology: Central dogma, gene expression
    + Technology: Measuring gene expression: marrays, RNAseq
    + Bioinformatics: Databases, R, Bioconductor
- Annotation databases
    + Feature centric
    + Sequence centric and genome browsers
- The Gene Ontology
    + What's the Go
    + What's in the GO
    + Browsing the GO
    + GO limitations
- Biological significance analysis using the GO
    + Using the GO to interpret our results
    + Why querying the GO is not enough. Overrepresentation analysis
    + From Overrepresentation analysis to GeneSet Expression Analysis


        
        
    

